api.go

```diff
@@ -50,6 +50,20 @@ var (
 	ErrCantBootstrap = errors.New("bootstrap only works on new clusters")
 )
 
+type Logger interface {
```

Comment:
What you propose would require anyone who wants to use a third party logger to implement a wrapper around it rather than just support the standard logger interface.

The way the original merger suggests, anyone who has a logger that supports `.Printf` (ie: like just about every logging library on the planet) can substitute their logging library of preference, and not have to build some interface layer between the proposed `.Output` and whatever they want to use?

The interface proposed is the standard one suggested by logrus:
https://github.com/sirupsen/logrus/blob/master/logrus.go#L97

And while you could make it private, why do you need to make it a secret? Doing so would preclude any external compile-time enforcement of the interface through for example:

```golang
type myLogger struct {
}
var _ raft.Logger = (*myLogger)(nil)
```

Reply:
further more if logrus had gone the Output route they could just provide a convient helper method that converts any logger that implements Output into their interface:

```golang
Adapt(l output) AdaptedLogger { ... }
```

Reply:
oh man logrus is even worse than i thought... doesnt tell you what line the log occurred on....